### PR TITLE
Fixes #15615: Exclude test files from godep reproducibility check

### LINKED ISSUE #15615: Updating dependencies fails on `*_test.go` files
In #15612 failures on shippable & travis are related to missing `*_test.go` files. This godep bump was done via `godep update github.com/google/cadvisor/...`. godep functionality is documented to not include `*_test.go` files & `testdata` directories.

I'd like to update `hack/verify-godeps.sh` to ignore these files in comparisons to make godep updates easier.

### PR DESCRIPTION
**Pull request by jimmidyson**

### CODE REVIEW

**Comment by jimmidyson** [comment]
Shippable failure is an unrelated flake.

File: hack/verify-godeps.sh [hunk 1]
```diff
@@ -92,7 +92,9 @@ fi
 # Godeps/_workstapces/src/github.com/fsouza/go-dockerclient/testing/data/symlink'
 # is an intentionally broken symlink. Linux can use --no-dereference. OS X cannot.
 # So we --exclude='symlink' so diff -r doesn't die following a bad symlink.
-if ! _out="$(diff -Naupr --exclude='symlink' ${KUBE_ROOT}/Godeps/_workspace/src ${_kubetmp}/Godeps/_workspace/src)"; then
+# godep doesn't include test files by default or when using `godep update` so
```
Comment (ixdy) [inline hack/verify-godeps.sh]: I think we should leave this as-is, but maybe add an error message advising to pass -t to godep if tests are the only thing broken?

Comment (jimmidyson) [inline hack/verify-godeps.sh]: `-t` only works with `godep save` so if you do a `godep update`, as I did for the cadvisor bump, test files are deleted. Why would you want to keep test files & test data vendored anyway?

Comment (jimmidyson) [inline hack/verify-godeps.sh]: @ixdy Any more thoughts?

**Comment by eparis** [comment]
I believe #15707 makes this irrelevant.

**Comment by jimmidyson** [comment]
Agreed.

**Pull request closed without merging**